packages/client/src/views/RecordAudit/ActionMenu.tsx

```diff
@@ -0,0 +1,646 @@
+
+  return (
+    isPrintable &&
+    userHasPrintScope && (
+      <DropdownMenu.Item
+        onClick={() => {
+          clearCorrectionAndPrintChanges(declarationId as string)
+          goToPrintCertificate(
+            declarationId as string,
+            (type as string).toLocaleLowerCase()
+          )
+        }}
+        disabled={!isDownloaded}
+      >
+        <Icon name="Printer" color="currentColor" size="large" />
+        {intl.formatMessage(messages.printDeclaration)}
+      </DropdownMenu.Item>
+    )
+  )
+}
+
+const IssueAction: React.FC<IActionItemCommonProps & IDeclarationProps> = ({
+  declarationId,
+  isDownloaded,
+  declarationStatus,
+  scope,
+  intl
+}) => {
+  const dispatch = useDispatch()
+
+  const isCertified = declarationStatus === SUBMISSION_STATUS.CERTIFIED
+
+  // @ToDo use: `record.print-issue-certified-copies` or other appropriate scope after configurable role pr is merged
+  const userHasIssueScope =
+    scope &&
+    ((scope as any as string[]).includes('register') ||
+      (scope as any as string[]).includes('validate'))
+
+  return (
+    isCertified &&
+    userHasIssueScope && (
+      <DropdownMenu.Item
+        onClick={() => {
+          dispatch(clearCorrectionAndPrintChanges(declarationId as string))
+          dispatch(goToIssueCertificate(declarationId as string))
+        }}
+        disabled={!isDownloaded}
+      >
+        <Icon name="Handshake" color="currentColor" size="large" />
+        {intl.formatMessage(messages.issueCertificate)}
+      </DropdownMenu.Item>
+    )
+  )
+}
+
+const DeleteAction: React.FC<{
+  intl: IntlShape
+  handleDelete: () => void
+  declarationStatus?: string
+}> = ({ intl, handleDelete, declarationStatus }) =>
+  declarationStatus === SUBMISSION_STATUS.DRAFT && (
+    <DropdownMenu.Item onClick={handleDelete}>
+      <Icon name="Trash" color="currentColor" size="large" />
+      {intl.formatMessage(buttonMessages.deleteDeclaration)}
+    </DropdownMenu.Item>
+  )
+const UnassignAction: React.FC<{
+  intl: IntlShape
+  handleUnassign: () => void
+  isDownloaded: boolean
+  assignment?: GQLAssignmentData
+  scope: Scope
+}> = ({ intl, handleUnassign, isDownloaded, assignment, scope }) => {
+  const isAssignedToSomeoneElse = !isDownloaded && assignment
+
+  // @ToDo use: appropriate scope after configurable role pr is merged
+  const userHasUnassignScope =
+    scope && (scope as any as string[]).includes('register')
+
+  return (
+    ((isAssignedToSomeoneElse && userHasUnassignScope) || isDownloaded) && (
+      <DropdownMenu.Item onClick={handleUnassign}>
+        <Icon name="ArrowCircleDown" color="currentColor" size="large" />
+        {intl.formatMessage(buttonMessages.unassign)}
+      </DropdownMenu.Item>
+    )
+  )
+}
+
+const UnassignModal: React.FC<{
+  intl: IntlShape
+  close: (result: boolean | null) => void
+  assignment?: GQLAssignmentData
+  isDownloaded: boolean
+}> = ({ intl, close, assignment, isDownloaded }) =>
+  assignment && (
```

Comment:
```suggestion

```

Let's have the calling end figure if the modal should be rendered to keep the component purer and move the conditional logic away from the view-components.